.github/workflows/differential-shellcheck.yml

```diff
@@ -0,0 +1,26 @@
+name: Differential ShellCheck
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    permissions:
+      security-events: write
+      pull-requests: write
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Differential ShellCheck
+        uses: redhat-plumbers-in-action/differential-shellcheck@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
why? seems shady.

Reply:
Why shady? [Token](https://github.com/jamacku/differential-shellcheck#token) is used for uploading SARIF to GitHub. The thing with `GITHUB_TOKEN` is that some actions use it as a default value, so you don't even know that action uses it. That seems a bit shady to me. But I understand your concerns.

Feel free to inspect [sources of action](https://github.com/redhat-plumbers-in-action/differential-shellcheck/tree/main/src), and I'm happy to answer your questions. Also, you can have a look at some repositories that are already using this action (e.g. [systemd](https://github.com/systemd/systemd/actions/workflows/differential-shellcheck.yml)).